Poll Configuration

VictoriaMalouf opened the pull request:
## PR type

- [ ] Feature
- [X] Bug Fix
- [X] Refactor
- [ ] Performance Optimization
- [ ] Documentation Update

## Description
<!--
A description of what this PR is, not how it does what it is supposed to do.
-->

- Added an Election form 
- Updated the validation of start/end dates/times
- Made new tests with refactored code
- Updated Election class to have 2 DateTime objects instead of 4 (start date, start time, end date, end time)

## Related Issues/Tickets
<!--
If this PR is related to a ticket/issue, include it here.
For github issues the format is as follows: "closes #1234" which would automatically close the github issue
For Jira, link the ticket.
-->
- Jira ticket:
- Related Issue #
- Closes #24
## QA Instruction, Screenshots etc...
<!--
Information on how to run/test the changes in the PR.
Include expected/actual results here as well.
-->
1. Run VotingSystemApplication
2. Go to http://localhost:8080/create-election
3. Attempt any invalid poll configuration to see that the correct error message is displayed

Examples include:
<img width="827" alt="Screen Shot 2024-11-24 at 12 09 46 PM" src="https://github.com/user-attachments/assets/1fb9b1d6-d49c-4a88-a108-03fcd38c7ada">

## Added/updated tests?

- [X] Yes
- [ ] No: _replace this line with details on why tests are not included in this PR_

## PR Checklist

- [X] I have rebased this branch against develop/latest
- [X] All the tests are passing
- [X] My commit messages are descriptive and usefull

its-jasmine reviewed (commented):
There's no feedback to the user when they don't provide a keys file at all.

VictoriaMalouf commented:
> There's no feedback to the user when they don't provide a keys file at all. Also no feedback when theres only one candidate provided.

I do have a check for # of candidates:

![Screen Shot 2025-01-12 at 1 51 13 PM](https://github.com/user-attachments/assets/63313bf3-aa17-43f5-9ea0-c1b63ddfbe88)

VictoriaMalouf commented:
Addressed having no file provided:
![Screen Shot 2025-01-12 at 2 49 47 PM](https://github.com/user-attachments/assets/5fb7875a-0238-4071-a462-0b99226ffeef)

Added case for empty file/no valid keys:
![Screen Shot 2025-01-12 at 2 50 09 PM](https://github.com/user-attachments/assets/27d5cad9-461e-4d8b-a623-bddf27ac8c6e)

its-jasmine commented:
> > There's no feedback to the user when they don't provide a keys file at all. Also no feedback when theres only one candidate provided.
> 
> I do have a check for # of candidates:
> 
> ![Screen Shot 2025-01-12 at 1 51 13 PM](https://private-user-images.githubusercontent.com/84146479/402379596-63313bf3-aa17-43f5-9ea0-c1b63ddfbe88.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzY3MzQyNDksIm5iZiI6MTczNjczMzk0OSwicGF0aCI6Ii84NDE0NjQ3OS80MDIzNzk1OTYtNjMzMTNiZjMtYWExNy00M2Y1LTllYTAtYzFiNjNkZGZiZTg4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTEzVDAyMDU0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTExY2ZhNmY3YzVhZmYzZmI3MmQwYmJiYmQxYTAyNjUyMTZmZTk4N2U4M2M1Yzk5MGFjZGI1NDViOTJiODg5NGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.vbvVzXXR8_Ub83s35MaYLOczqLeIH8puHkkGk71qsjI)

If you try putting a word that is more than one letter long it doesn't show the message.
This worked:
![image](https://github.com/user-attachments/assets/c6e9c7ee-4779-4ac9-8c41-6828145c693a)

But this didn't:
![image](https://github.com/user-attachments/assets/94edd4b8-8545-43e8-9168-eabd505aaffe)
